brokerapi/brokers/cloudsql/sql_account_manager.go

```diff
@@ -0,0 +1,203 @@
+		return nil, fmt.Errorf("Error encountered waiting for operation %q to finish: %s", op.Name, err)
+	}
+
+	return &sqlUserAccount{
+		Username: request.Name,
+		Password: request.Password,
+	}, nil
+}
+
+func (broker *CloudSQLBroker) deleteSqlUserAccount(ctx context.Context, binding models.ServiceBindingCredentials, instance models.ServiceInstanceDetails) error {
+	var creds sqlUserAccount
+	if err := json.Unmarshal([]byte(binding.OtherDetails), &creds); err != nil {
+		return fmt.Errorf("Error unmarshalling credentials: %s", err)
+	}
+
+	client, err := broker.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	op, err := client.Users.Delete(broker.ProjectId, instance.Name, "", creds.Username).Do()
+	if err != nil {
+		return fmt.Errorf("Error deleting user: %s", err)
+	}
+
+	if err := broker.pollOperationUntilDone(ctx, op, broker.ProjectId); err != nil {
+		return fmt.Errorf("Error encountered waiting for operation %q to finish: %s", op.Name, err)
+	}
+
+	return nil
+}
+
+func (broker *CloudSQLBroker) createSqlSslCert(ctx context.Context, vars *varcontext.VarContext, instance models.ServiceInstanceDetails) (*sqlSslCert, error) {
+	request := &googlecloudsql.SslCertsInsertRequest{
+		CommonName: vars.GetString("certname"),
+	}
+
+	if err := vars.Error(); err != nil {
+		return nil, err
+	}
+
+	// create username, pw with grants
+	client, err := broker.createClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	newCert, err := client.SslCerts.Insert(broker.ProjectId, instance.Name, request).Do()
+	if err != nil {
+		return nil, fmt.Errorf("Error creating SSL certs: %s", err)
+	}
+
+	// poll for the user creation operation to be completed
+	return &sqlSslCert{
+		Sha1Fingerprint: newCert.ClientCert.CertInfo.Sha1Fingerprint,
+		CaCert:          newCert.ServerCaCert.Cert,
+		ClientCert:      newCert.ClientCert.CertInfo.Cert,
+		ClientKey:       newCert.ClientCert.CertPrivateKey,
+	}, nil
+}
+
+func (broker *CloudSQLBroker) deleteSqlSslCert(ctx context.Context, binding models.ServiceBindingCredentials, instance models.ServiceInstanceDetails) error {
+	var creds sqlSslCert
+	if err := json.Unmarshal([]byte(binding.OtherDetails), &creds); err != nil {
+		return fmt.Errorf("Error unmarshalling credentials: %s", err)
+	}
+
+	client, err := broker.createClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// If we didn't generate SSL certs for this binding, then we cannot delete them
+	if creds.CaCert == "" {
+		return nil
+	}
+
+	op, err := client.SslCerts.Delete(broker.ProjectId, instance.Name, creds.Sha1Fingerprint).Do()
+	if err != nil {
+		return fmt.Errorf("Error deleting ssl cert: %s", err)
+	}
+
+	if err := broker.pollOperationUntilDone(ctx, op, broker.ProjectId); err != nil {
+		return fmt.Errorf("Error encountered waiting for operation %q to finish: %s", op.Name, err)
+	}
+
+	return nil
+}
+
+func toJsonMap(obj interface{}) (map[string]interface{}, error) {
```

Comment:
It seems like toJsonMap and mergeJsonProperties are orthogonal to the SQL broker's responsibilities. Consider moving them to a separate library file and adding unit tests so that other parts of the system can benefit from their utility.